### ISSUE TITLE
The multiple bundles workflows seems like it's missing a common use-case.

### ISSUE THREAD

**Issue by philipwalton** [browserify/browserify#928]
The current [multiple bundles example](https://github.com/substack/node-browserify#multiple-bundles) seems to be mainly optimizing for the use-case of excluding common core dependencies (e.g. jQuery or LoDash).

It doesn't seem to make it very easy to generically exclude all modules from a bundle that I know to already be included on the page. At least not without a lot of manual config.

Here's the use-case:
- You have a site with some core JavaScript code that runs on every page.
  - That code depends on modules which may depend on other modules which may depend on other modules, etc.
- You have page-specific JavaScript that is only loaded on certain pages.
  - The page-specific JavaScript very likely depends on some of the same modules that the core depends on, so it shouldn't re-bundle any of that.
  - The page specific JavaScript does not know, exhaustively, what modules core depends on, all it knows is that core is already loaded on the page.

Ideally, I should be able to build a page specific bundle and tell it to exclude (or consider external) any of the modules already bundled in some other bundle. For example:

```
$ browserify --require-all core > static/core.js
$ browserify --require-all --external-deep ./core page.js > static/page.js
```

Basically the `--require-all` flags would tell browserify to expose all modules globally, and the `--external-deep` flags would tell browserify not to bundle any modules in the dependency graph of `./core` when making `static/page.js`

Though the names are a little clunky, this should meet all of the requirements for the use-case I described.
